index.js

```diff
@@ -84,6 +84,13 @@ function parse(str, options) {
  */
 
 function serialize(name, val, options) {
+  if (typeof name !== 'string') {
+    throw new TypeError('argument name must be a string');
+  }
+  if (typeof val !== 'string') {
```

Comment:
The current encoder works just fine with numbers, Dates, etc. so it would still be a breaking change.